lib/std/builtin.zig

```diff
@@ -47,25 +47,25 @@ pub const subsystem: ?SubSystem = blk: {
     }
 };
 
-/// This data structure is used by the Zig language code generation and
-/// therefore must be kept in sync with the compiler implementation.
```

Comment:
What's wrong with this being part of the doc comments? I think it's interesting and useful to know this when using these types.

Reply:
This feels like a message to the code's maintainer. Documentation for users explains how to use a thing, and these explain why it's implemented the way it is. I would expect the doc comments for `StackTrace`, for example, to explain how to get one and maybe something about the order of the instruction_addresses.

Reply:
I don't think I made my case well there. I agree that it's interesting, but disagree about useful.

What actions should a user take in order to accommodate the fact that the data structure is used by the implementation and must be kept in sync with the compiler implementation? I don't see it. I could see maybe these being implementation dependent once we have multiple implementations, but that feels like the kind of thing we should standardize.

The closest to useful I can see is that since the compiler implementation is in flux, people might need to change code that uses these data structures more often than others, but that's broadly true for the standard library anyway. And by the time the standard library stabilizes I expect the compiler will already be fairly stable.

If that wasn't convincing, I hope you'll explain why. I've seen several other examples of doc comments that I really think ought to be regular comments, and it's important to understand before I go changing those.

Reply:
Understanding the purpose of the existence of a struct helps the user to understand how it is governed by semantic versioning of the library it is contained in, and under what conditions the struct fields would be changed. Also, it's documentation that is relevant to the struct. What's the point of a comment on a struct if we're going to hide it from being exposed as documentation for that struct?

This feels a bit bike sheddy to me and I'd rather spend my limited energy discussing other things, but I do have a strong opinion here. So I'm sorry to do this but I'm going to pull rank on this one.